QuietFactory.php

```diff
@@ -69,10 +71,28 @@ public function createItem($name, array $options = array())
             }
 
             // remove route and content options
-            unset($options['route']);
-            unset($options['content']);
+            unset($options['route'], $options['content']);
 
             return $this->innerFactory->createItem($name, $options);
         }
     }
+
+    /**
+     * Forward adding extensions to the wrapped factory.
+     *
+     * @param ExtensionInterface $extension
+     * @param int                $priority
+     *
+     * @throws \Exception if the inner factory does not implement the addExtension method.
+     */
+    public function addExtension(ExtensionInterface $extension, $priority = 0)
+    {
+        if (!method_exists($this->innerFactory, 'addExtension')) {
+            throw new LogicException(sprintf(
```

Comment:
this has a tiny potential of starting to break for people that somehow relied on this not to work properly...